src/runtime/gc_custom.go

```diff
@@ -0,0 +1,49 @@
+//go:build gc.custom
+// +build gc.custom
+
+package runtime
+
+// This GC strategy allows an external GC to be plugged in instead of the builtin
+// implementations.
+//
+// The custom implementation must provide the following functions in the runtime package
+// using the go:linkname directive:
+//
+// - func initHeap()
+// - func alloc(size uintptr, layout unsafe.Pointer) unsafe.Pointer
+// - func free(ptr unsafe.Pointer)
+// - func GC()
+// - func KeepAlive(x interface{})
+// - func SetFinalizer(obj interface{}, finalizer interface{})
```

Comment:
See https://github.com/tinygo-org/tinygo/pull/3336. We don't support these yet so I've simplified things a bit for now.

Reply:
It's not a high priority but I was going to implement setFinalizer in the custom implementation. Is it ok to keep them to try it? I guess in principle anyways a finalizer is a feature of the gc

Reply:
I'm interested in implementing finalizer, but for now I've removed the methods from this PR to make sure it compiles with the dev branch where the change was merged